### ISSUE TITLE
Testing current master with lethekit

### ISSUE THREAD

**Issue by gorazdko** [BlockchainCommons/bc-keytool-cli#5]
### Test vector 1
```bash
seed_lethekit=$(seedtool --in ur ur:crypto-seed/oyadgdvlpfssfwmkztcebbnyzowkspnljlrhdkztykahhf)

keytool \
    --seed $seed_lethekit \
    --address-index '*' \
    output-descriptor
```

result OK, matches with lethekit:
```bash
wpkh([b238a845/84h/0h/0h]xpub6D1GQL3S3qryEx5gQccdVB5Fy1ePkeDGU8bMmW1GKfDQvtBsbJ6FaaudsBphw2WwtjYyJ4ARR6Mh9ZRTTKqro6eZww2m6SVxG7WuY2DALVp/0/*)
```
![descriptor_wpkh](https://user-images.githubusercontent.com/25270775/97021685-712a5280-1553-11eb-8d55-b12453b61da1.png)

### Test vector 2

```bash
# address wpkh not supported only pkh and sh?
for (( i=0; i <= 9; ++i ))
do
    keytool \
    	--seed 1eb5338edac6fae54cbb172091ae6c1a \
    	--address-index $i \
    	address-index \
    	full-address-derivation-path \
    	address-pkh
done
```

Result:
```bash
0
c1859926/84h/0h/0h/0/0
17u4HStkGWxJ5PtNqzCMEKRzaPJ8h35eBc
1
c1859926/84h/0h/0h/0/1
1JyxrbHYS7ZGhq68JewhJAMWka38Bv8tFw
2
```

Here I noticed no support for segwit addresses. Also, the output shows segwit derivation path (84h) when `address-pkh` specified (`44h`) -> Not OK.

**Comment by wolfmcnally** [comment]
@gorazdko I think I had some previous conversations with @ChristopherA and/or @Fonta1n3 where they thought the best default value for the `purpose` path component was 84. Please advise.

**Comment by ChristopherA** [comment]
@wolfmcnally I think that came from the discussion that if the output is a complete descriptor (which is what keytool-cli does), and as LetheKit can only do single signature, the default should be 84' crypto-hdkey as that is segwit or its SLIP-132 form of zpub/zprv (I hate SLIP-132, but as future Gordian will not be supporting singlesig except to sweep, and that is what Ledger/Trezor use for segwit, we have to continue to offer it as an option.)

However, we want to focus with LetheKit export not of descriptors but of the crypto-hdkey that is a cosigner, i.e. `[48'/0'/0'/2']xpub…` and the cosigner's private key `[48'/0'/0'/2']xprv…`.  These two should be the minimal number of clicks, then the singlesig segwit zpub/zprv, then everything else can have more manual entry steps in LetheKit.

Similarly, we need these in seedtool to be the defaults. If you ask for descriptor and don't define a path, it defaults to /84'/0'/0'/0'. If you ask for SLIP-132 and don't offer path offers zpub/zprv form, if you ask for hdkey and  don't define path, you get `[48'/0'/0'/2']xpub…`.

**Comment by gorazdko** [comment]
> I think I had some previous conversations with @ChristopherA and/or @Fonta1n3 where they thought the best default value for the purpose path component was 84. Please advise.

Yes I agree 84 to be default, but there are flags that should change the default value of purpose. Just like [here](https://github.com/BlockchainCommons/bc-keytool-cli/commit/7604c8089008896bf86c7c18c800aeb86fe5f83e) where you made  `purpose` dependent on `output-type`.

The flags that should change the default purpose are those which hold values: pkh, wpkh, sh-wpkh

The result I pasted above:

```bash
c1859926/84h/0h/0h/0/0
17u4HStkGWxJ5PtNqzCMEKRzaPJ8h35eBc
```
is not wrong, it looks weird though, because 84h is standard for segwit addresses (bech32) and 44h is standard for pkh addresses (base58check)


@wolfmcnally also please test my lethekit [PR](https://github.com/BlockchainCommons/bc-lethekit/pull/60) with keytool and Guardian to confirm that my implementation of cosigner is correct.
